src/overlays/actors/ovl_Obj_Tsubo/z_obj_tsubo.c

```diff
@@ -46,39 +73,275 @@ static ColliderCylinderInit D_80BA1B94 = {
+    }
+
+    func_80033480(globalCtx, &this->actor.world.pos, 30.0f, 4, 0x14, 0x32, 1);
+}
+
+void ObjTsubo_WaterBreak(ObjTsubo* this, GlobalContext* globalCtx) {
+    s32 pad[2];
+    s16 angle;
+    Vec3f pos = this->actor.world.pos;
+    Vec3f velocity;
+    s32 phi_s0;
+    s32 i;
+
+    pos.y += this->actor.yDistToWater;
+    EffectSsGSplash_Spawn(globalCtx, &pos, NULL, NULL, 0, 400);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Obj_Tsubo/func_80BA0DF4.s")
+    for (i = 0, angle = 0; i < 15; i++, angle += 0x4E20) {
+        f32 sins = Math_SinS(angle);
+        f32 coss = Math_CosS(angle);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Obj_Tsubo/func_80BA0E98.s")
+        pos.x = sins * 8.0f;
+        pos.y = (Rand_ZeroOne() * 5.0f) + 2.0f;
+        pos.z = coss * 8.0f;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Obj_Tsubo/ObjTsubo_Init.s")
+        velocity.x = pos.x * 0.2f;
+        velocity.y = (Rand_ZeroOne() * 4.0f) + 2.0f;
+        velocity.z = pos.z * 0.2f;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Obj_Tsubo/ObjTsubo_Destroy.s")
+        Math_Vec3f_Sum(&pos, &this->actor.world.pos, &pos);
+        phi_s0 = (Rand_ZeroOne() < .2f) ? 64 : 32;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Obj_Tsubo/func_80BA100C.s")
+        EffectSsKakera_Spawn(globalCtx, &pos, &velocity, &this->actor.world.pos, -180, phi_s0, 30, 30, 0,
+                             (Rand_ZeroOne() * 95.0f) + 15.0f, 0, 32, 70, KAKERA_COLOR_NONE,
+                             sObjectIds[(this->actor.params >> 8) & 1], D_80BA1B8C[(this->actor.params >> 8) & 1]);
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Obj_Tsubo/func_80BA1294.s")
+void func_80BA152C(ObjTsubo* this) {
+    this->actionFunc = func_80BA153C;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Obj_Tsubo/func_80BA152C.s")
+void func_80BA153C(ObjTsubo* this, GlobalContext* globalCtx) {
+    if (Object_IsLoaded(&globalCtx->objectCtx, this->objTsuboBankIndex)) {
+        this->actor.draw = ObjTsubo_Draw;
+        this->actor.objBankIndex = this->objTsuboBankIndex;
+        ObjTsubo_SetupIdle(this);
+        this->actor.flags &= ~0x10;
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Obj_Tsubo/func_80BA153C.s")
+void ObjTsubo_SetupIdle(ObjTsubo* this) {
+    this->actionFunc = ObjTsubo_Idle;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Obj_Tsubo/func_80BA15AC.s")
+void ObjTsubo_Idle(ObjTsubo* this, GlobalContext* globalCtx) {
+    s32 pad;
+    s16 temp_v0;
+    s32 phi_v1;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Obj_Tsubo/func_80BA15BC.s")
+    if (Actor_HasParent(&this->actor, globalCtx)) {
+        ObjTsubo_SetupLiftedUp(this);
+        return;
+    }
+    if ((this->actor.bgCheckFlags & 0x20) && (this->actor.yDistToWater > 15.0f)) {
```

Comment:
cases can be made for either side here, but I think this would read nicer with else ifs instead of early returns